lib/dry/system/container.rb

```diff
@@ -216,7 +244,22 @@ def import(other)
         # @return [self]
         #
         # @api public
-        def boot(name, opts = {}, &block)
+        def boot(name, **opts, &block)
+          boot_condition = opts[:if]
+
+          if boot_condition
+            begin
+              case boot_condition
+              when Symbol
+                return unless __send__(boot_condition)
+              else
+                return unless boot_condition.call
+              end
+            rescue NoMethodError => e
```

Comment:
Are you intentionally wrapping both cases (boot_condition being either a Symbol or a proc) in this begin/rescue? Wouldn't it be better just to wrap the `boot_condition.call` case?

Reply:
hey, the general idea here: user can use bool value:

```ruby
MyApp.boot(:db, if: false) do |container|
  # ...
end
```

In this case, the user gets `NoMethodError ` and it will be bad from UX perspective. Instead of this, I suggest to raise a normal exception and say that a user can't use this object as an argument. WDYT?

Reply:
@davydovanton Yeah, I think this'd be the better way to go

Reply:
@davydovanton I wonder if we can actually expand the range of values we accept a little...

Firstly, given a `String` or `Symbol` object — make it so _both_ of these are considered container identifiers, and we resolve the matching object from the container accordingly.

Then, by this point, we should have either a:

(1) Callable object, or
(2) Something that's otherwise truthy or falsy, per standard Ruby behaviour

So by this point here we just .call() if we can, otherwise rely on the truthy/falsy result of whatever expression we've been passed.

How does that feel?